js/gateio.js

```diff
@@ -648,15 +649,17 @@ module.exports = class gateio extends Exchange {
                     const baseId = this.safeString (parts, 0);
                     const quoteId = this.safeString (parts, 1);
                     const date = this.safeString (parts, 2);
-                    const linear = quoteId.toLowerCase () === settle;
-                    const inverse = baseId.toLowerCase () === settle;
                     const base = this.safeCurrencyCode (baseId);
                     const quote = this.safeCurrencyCode (quoteId);
+                    const settle = this.safeCurrencyCode (settleId);
+                    const linear = quote === settle;
+                    const inverse = base === settle;
+                    const expiry = parseInt (Precise.stringMul (this.safeString (market, 'expire_time'), '1000'));
```

Comment:
Let's use `this.safeTimestamp` here (and everywhere else in similar cases):

```JavaScript
const expiry = parseInt (Precise.stringMul (this.safeString (market, 'expire_time'), '1000'));
```

↓

```JavaScript
const expiry = this.safeTimestamp (market, 'expire_time');
```